### PR TITLE
Update to the latest `config`

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR updates the repository to use the latest `config`.

Mostly, this is done in order to pull in the following changes:

1. [Make published artifacts full-fledged #335](https://github.com/SpineEventEngine/config/pull/335).
2. [Bump dependencies #344](https://github.com/SpineEventEngine/config/pull/344).

Applying of those changes led to the following:

1. `testJar` artifact has disappeared from the list of artifacts published by the modules. This artifact is [optional](https://github.com/SpineEventEngine/config/pull/340) now and not published by default.
2. `main` and `test` source sets have been extended with the generated code. This enables presence of the generated sources in `sourcesJar` artifact.

Please note, the repository does not use the latest versions of Spine libraries and some of their dependencies. In particular `mc-java`, ErrorProne and Commons Cli. Take a look on #181 for details.

The library version is bumped to `2.0.0-SNAPSHOT.73`.

### CODE REVIEW

File: integration-tests/test-app/build.gradle.kts [hunk 1]
```diff
@@ -1,5 +1,3 @@
-import io.spine.internal.gradle.checkstyle.CheckStyleConfig
-
 /*
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply integration-tests/test-app/build.gradle.kts:1]: @sonatype-lift ignore

File: pom.xml [hunk 2]
```diff
@@ -74,19 +74,19 @@ all modules and does not describe the project structure per-subproject.
   <dependency>
     <groupId>com.google.protobuf</groupId>
     <artifactId>protobuf-java</artifactId>
-    <version>3.19.2</version>
+    <version>3.19.4</version>
     <scope>compile</scope>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:80]: @sonatype-lift ignore

File: pom.xml [hunk 3]
```diff
@@ -56,13 +56,13 @@ all modules and does not describe the project structure per-subproject.
...
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:56]: @sonatype-lift ignore

File: pom.xml [hunk 4]
```diff
@@ -56,13 +56,13 @@ all modules and does not describe the project structure per-subproject.
...
     <scope>compile</scope>
   </dependency>
   <dependency>
     <groupId>com.google.http-client</groupId>
     <artifactId>google-http-client</artifactId>
-    <version>1.40.1</version>
+    <version>1.41.5</version>
     <scope>compile</scope>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:68]: @sonatype-lift ignore

File: pom.xml [hunk 5]
```diff
@@ -56,13 +56,13 @@ all modules and does not describe the project structure per-subproject.
...
     <groupId>com.google.guava</groupId>
     <artifactId>guava</artifactId>
-    <version>31.0.1-jre</version>
+    <version>31.1-jre</version>
     <scope>compile</scope>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:62]: @sonatype-lift ignore

File: pom.xml [hunk 6]
```diff
@@ -156,12 +156,12 @@ all modules and does not describe the project structure per-subproject.
...
     <groupId>com.google.protobuf</groupId>
     <artifactId>protoc</artifactId>
-    <version>3.19.2</version>
+    <version>3.19.4</version>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:161]: @sonatype-lift ignore

**Comment by yevhenii-nadtochii** [comment]
@armiol Please take a look.

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**